pkg/controller/mizar/mizar-pod-controller.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2020 Authors of Arktos.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mizar
+
+import (
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+const (
+	controllerForMizarPod = "mizar_pod"
+)
+
+// MizarPodController points to current controller
+type MizarPodController struct {
+	kubeClient clientset.Interface
+
+	// A store of objects, populated by the shared informer passed to MizarPodController
+	lister corelisters.PodLister
+	// listerSynced returns true if the store has been synced at least once.
+	// Added as a member to the struct to allow injection for testing.
+	listerSynced cache.InformerSynced
+
+	// To allow injection for testing.
+	handler func(keyWithEventType KeyWithEventType) error
+
+	// Queue that used to hold thing to be handled.
+	queue workqueue.RateLimitingInterface
+
+	grpcHost string
+}
+
+// NewMizarPodController creates and configures a new controller instance
+func NewMizarPodController(informer coreinformers.PodInformer, kubeClient clientset.Interface, grpcHost string) *MizarPodController {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClient.CoreV1().EventsWithMultiTenancy(metav1.NamespaceAll, metav1.TenantAll)})
+
+	c := &MizarPodController{
+		kubeClient:   kubeClient,
+		lister:       informer.Lister(),
+		listerSynced: informer.Informer().HasSynced,
+		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerForMizarPod),
+		grpcHost:     grpcHost,
+	}
+
+	informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.createObj,
+		UpdateFunc: c.updateObj,
+		DeleteFunc: c.deleteObj,
+	})
+	c.lister = informer.Lister()
+	c.listerSynced = informer.Informer().HasSynced
+
+	c.handler = c.handle
```

Comment:
Could you change the naming here? It's kinda confusing

Reply:
+1, a common func in this case is sync(). Does it fit?